crates/viewer/re_space_view_dataframe/src/dataframe_ui.rs

```diff
@@ -0,0 +1,350 @@
+        let row_id_column_index = schema
+            .iter()
+            .find_position(|desc| match desc {
+                ColumnDescriptor::Control(control_column_desc) => {
+                    control_column_desc.component_name
+                        == ComponentName::from("rerun.controls.RowId")
+                }
+                _ => false,
+            })
+            .map(|(pos, _)| pos);
+
+        Ok(Self {
+            display_record_batches,
+            batch_and_row_from_row_nr,
+            time_column_index,
+            row_id_column_index,
+        })
+    }
+}
+
+/// [`egui_table::TableDelegate`] implementation for displaying a [`QueryHandle`] in a table.
+struct DataframeTableDelegate<'a> {
+    ctx: &'a ViewerContext<'a>,
+    query_handle: &'a QueryHandle<'a>,
+    schema: &'a [ColumnDescriptor],
+    header_entity_paths: Vec<Option<EntityPath>>,
+    display_data: anyhow::Result<RowsDisplayData>,
+
+    num_rows: u64,
+}
+
+impl<'a> egui_table::TableDelegate for DataframeTableDelegate<'a> {
+    fn prepare(&mut self, info: &egui_table::PrefetchInfo) {
+        re_tracing::profile_function!();
+
+        let data = RowsDisplayData::try_new(
+            &info.visible_rows,
+            self.query_handle.get(
+                info.visible_rows.start,
+                info.visible_rows.end - info.visible_rows.start,
+            ),
+            self.schema,
+            &self.query_handle.timeline(),
+        );
+
+        self.display_data = data.with_context(|| "Failed to create display data");
+    }
+
+    fn header_cell_ui(&mut self, ui: &mut egui::Ui, cell: &egui_table::HeaderCellInfo) {
+        egui::Frame::none()
+            .inner_margin(egui::Margin::symmetric(4.0, 0.0))
+            .show(ui, |ui| {
+                if cell.row_nr == 0 {
+                    if let Some(entity_path) = &self.header_entity_paths[cell.group_index] {
+                        ui.label(entity_path.to_string());
+                    }
+                } else if cell.row_nr == 1 {
+                    ui.strong(self.schema[cell.col_range.start].short_name());
+                } else {
+                    // this should never happen
+                    error_ui(ui, format!("Unexpected header row_nr: {}", cell.row_nr));
+                }
+            });
+    }
+
+    fn cell_ui(&mut self, ui: &mut egui::Ui, cell: &egui_table::CellInfo) {
+        re_tracing::profile_function!();
+
+        if cell.row_nr % 2 == 1 {
+            // Paint stripes
+            ui.painter()
+                .rect_filled(ui.max_rect(), 0.0, ui.visuals().faint_bg_color);
+        }
+
+        // sanity check, this should never happen
+        if cell.row_nr >= self.num_rows {
+            error_ui(
+                ui,
+                format!(
+                    "Unexpected row_nr: {} (table row count {})",
+                    cell.row_nr, self.num_rows
+                ),
+            );
+            return;
+        }
```

Comment:
Let's make sure we don't miss these bugs

```suggestion
    debug_assert!(cell.row_nr < self.num_rows, "Bug in egui_table");
```